main.py

```diff
@@ -148,5 +148,72 @@ async def keen(context):
     await context.send(result)
 
 
-bot.run(TOKEN)
+@bot.command(name='daily', help='Will give the daily leaderboard for specified data')
+async def daily(context, day):
+    # Only respond if used in a channel called 'advent-of-code'
+    if context.channel.name != 'advent-of-code':
+        return
+
+    players = get_players()
+
+    players_day = []
+
+    # Goes through all the players checking if they have data for that day and if they do adding to players_days
+    for player in players:
+        if day in player[4]:
+            players_day.append(player)
+            
+
+    #players_day has all people who have finished one star for that day
+    first_star = []
+    second_star = []
+
+    # Adds all the players which has 
+    for player_day in players_day:
+        if '1' in player_day[4][day]:
+            
+            first_star.append((player_day[0], int(player_day[4][day]['1']['get_star_ts'])))
+
+        if '2' in player_day[4][day]:
+            second_star.append((player_day[0], int(player_day[4][day]['2']['get_star_ts'])))
+    
+    #Sorts the two lists on timestamps.(Used insertion as lists aren't too large and easy to change to order on correct data)
+    first_star.sort(key=lambda data : data[1])
+    second_star.sort(key=lambda data : data[1])
 
+    final_table = []
+
+    #Adds all the people from first list
+    for i, player in enumerate(first_star):#
+        final_table.append((player[0], (len(players) - i)))
+    
+    
+
+    #Updates the list with all the people who got the second star and their score
+    for i, player in enumerate(second_star):
+        index = [i for i, item in enumerate(final_table) if item[0] == player[0]][0]
+        to_change = final_table[index]
+        final_table[index] = (to_change[0], (to_change[1] + (len(second_star) - i)))
+    
+    #Sorts the table
+    final_table.sort(reverse=True,key=lambda data : data[1])
+
+    print(final_table)
+
+    #Outputs data
+    result = ""
+    if not final_table:
+        result = "```No Scores for this day yet```"
+    else:
+        result = "```"
+
+        for place, player in enumerate(final_table):
+            result += str(place+1) + ": " + str(player[0]) + " with " + str(player[1]) +" point(s)\n"
```

Comment:
It would be nice if this format could:
A. Use str.format() instead of concatenating a bunch of strings together, and
B. Use the same output format as is used for the other commands, the format string is available on line 18 and there are a couple of examples where you can see how it's used in the code

Reply:
Sorted